### ISSUE TITLE
Adding customField into PDF

### ISSUE THREAD

**Issue by ppiuhola** [bytefury/crater#409]
I'm strugling to add "custom field" into PDF file, what is the syntax to call it correctly?
I noticed, it's not using the same format than for invoice email in customization settings like {CUSTOM_INVOICE_IBAN_CODE}.
I'll be happy to any hint/advice to solve this out.

User guide is not good enough level regarding custom values.

**Comment by mwlandolsi** [comment]
Intersted

**Comment by TheZoker** [comment]
Here is where the custom fields are handled:
https://github.com/bytefury/crater/blob/99c27ba8ef8f498c6912556b9625ffbd16719632/app/Traits/GeneratesPdfTrait.php#L155-L161

So it uses the custom fields slug. The slug is created the following way:
https://github.com/bytefury/crater/blob/99c27ba8ef8f498c6912556b9625ffbd16719632/app/Models/CustomField.php#L116

And again:
https://github.com/bytefury/crater/blob/99c27ba8ef8f498c6912556b9625ffbd16719632/app/Space/helpers.php#L103-L106

Long story short:
`{CUSTOM_<model>_<label>}`

`<model>` = the place where the custom field is shown (`INVOICE`, `CUSTOMER` etc.) 
`<label>` = the label of the field

So a field that is shown on the invoice with the name `Taxnumber` should be `{CUSTOM_INVOICE_TAXNUMBER}`

I don't see any other custom field handling within the code. So this should be the right way.

**Comment by DRogueRonin** [comment]
@ppiuhola 

I'm using
```
<p>{{ $invoice->getFormattedString('{CUSTOM_INVOICE_IBAN_CODE}') }}</p>
```

the `getFormattedString` function is converting the placeholder to the custom field value.

**Comment by mohitpanjwani** [comment]
Hey! I'll be adding some documentation for this soon but this is how you can use a custom field: 

For custom fields added to Invoice: 
{{$invoice->user->getCustomFieldBySlug('CUSTOM_CUSTOMER_GST')->default_answer}}


For custom field added to customer: 
{{$invoice->user->getCustomFieldBySlug('CUSTOM_CUSTOMER_GST')->default_answer}}

I just noticed that we are not displaying the Slug field on the Custom Fields table UI so its easier to copy but I'll work on adding that. 

Meanwhile you can copy the slug from the database. The format for slug is CUSTOM_{MODEL}_{NAME}

**Comment by mohitpanjwani** [comment]
If you would like to use the custom field's label or any other field then you can do that as well: 

```
<tr>
    <td class="attribute-label">{{$invoice->user->getCustomFieldBySlug('CUSTOM_CUSTOMER_GST')->customField->label}}</td>
    <td class="attribute-value"> {{$invoice->user->getCustomFieldBySlug('CUSTOM_CUSTOMER_GST')->default_answer}}</td>
</tr>
```

**Comment by perteraul** [comment]
I have created a custom invoice, that is being shared by two of my Crater companies. In each of the company settings, I have created a custom invoice field, called `Project Name` and that has the same settings, however, the unique slug is different. 

How can I make it so that it works with both `CUSTOM_INVOICE_PROJECT` and `CUSTOM_INVOICE_PROJECT_1`?

`<td class="attribute-label">{{$invoice->user->getCustomFieldBySlug('CUSTOM_INVOICE_PROJECT')->customField->placeholder}}</td>`

Thanks for the help!

**Comment by NicoHood** [comment]
In the crater settings we define (for example):
```
{COMPANY_NAME}
{COMPANY_ADDRESS_STREET_1}
{COMPANY_ADDRESS_STREET_2}
{COMPANY_ZIP_CODE} {COMPANY_CITY}
{SHIPPING_COUNTRY}
```

But how can we access those via the html template? `{{ $company_name }}` Did not work

**Comment by enoversum** [comment]
> But how can we access those via the html template? `{{ $company_name }}` Did not work

I just found out how, just like this:

`{{ $invoice->getFormattedString('{COMPANY_PHONE}') }}`

**Comment by fgiering** [comment]
> Hey! I'll be adding some documentation for this soon but this is how you can use a custom field:

Unfortunately its still not in the docs.
I tried with `getFormattedString()` but then HTML-Tags are in the plain String to. How to get line-breaks etc. from a customfield which is a textarea correctly?

**Comment by albertvisuals** [comment]
I have tried to add: {{$invoice->user->getCustomFieldBySlug('CUSTOM_CUSTOMER_UID_NUMMER')->default_answer}}
But it returns an error: 
Error
Call to a member function getCustomFieldBySlug() on null (View: /XXX/resources/views/app/pdf/invoice/FOTOHOF.blade.php)

 {{$invoice->getCustomFieldBySlug('CUSTOM_INVOICE_MITARBEITERIN')->default_answer}} is working though....
